### ISSUE TITLE
Building an end-to-end test for ReinforceNormal

### ISSUE THREAD

**Issue by ssampang** [Element-Research/dpnn#52]
My model is still failing to learn, so to be sure that my ReinforceGamma module is doing backprop correctly, I'd like to build a simple test to prove that it can learn correctly, and to ensure that the test works, I wanted to run it on the ReinforceNormal module first. I was thinking something along the lines of regression where a Linear module placed before the Reinforce module would learn to scale the input by 100:

```
net = nn.ConcatTable()
net:add( nn.Sequential():add( nn.Linear(1,1) ):add( nn.ReinforceNormal(30) )
net:add( nn.Sequential():add( nn.Constant(1,1) ):add( nn.Add(1) )

crit = nn.VRClassReward(net,1,nil,reward)

function gradUpdate(asd, x, y, criterion, learningRate)
  pred = asd:forward(x)
  err = criterion:forward(pred,y)
  gradCriterion = criterion:backward(pred,y)
  asd:zeroGradParameters()
  asd:backward(x,gradCriterion)
  asd:updateParameters(learningRate)
  return err
end

for i=1,10 do
  x = torch.rand(1)
  y = torch.mul(x,100)
  err = gradUpdate(q,x,y,c,1.0)
end
```

where the VRClassReward module is modified to accept a function on initialization, which is used to calculate the reward given an input and a target. Specifically, [this](https://gist.github.com/ssampang/3f3edd816849b1f539a4e763659c29b0) is my modified VRClassReward, and this is my reward function:

```
-- this function should just return 1 if x and y are within 10 of each other and 0 otherwise
function reward(x,y)                                   
  local z = torch.abs( torch.add(x:float(),-1,y:float())):apply( function(q) if q < 10 then return 1 else return 0 end end)
  return z
end
```

Ideally, I expected this to allow `net.modules[1].modules[1].weight` to converge to 100, or somewhere close, but I'm seeing large negative values, and I can't figure out why...any suggestions?

**Comment by nicholas-leonard** [comment]
I think this is solved right?

**Comment by ssampang** [comment]
Yep, forgot to close it.
